data/scripts/dependency-resolver.py

```diff
@@ -144,40 +205,125 @@ def handle_python_check(args, conf):
 
     cmd = "%s %s" % (sys.executable, f.name)
     output, status = run_command(cmd)
-    makefile_vars["HAVE_PYTHON_%s" % dep] = "y" if status else "n"
+    context.add_cond_makefile_var("HAVE_PYTHON_%s" % dep, "y" if status else "n")
+
+def handle_cflags_check(args, conf, context):
+    check_cflags = conf.get("cflags")
+    append_to = conf.get("append_to")
+    source = cstub.format(headers="", fragment="")
+
+    if not append_to or not check_cflags:
+        return
+
+    wanted = []
+    for i in check_cflags:
+        wanted.append(i)
+
+    flags = " ".join(wanted)
+    success = compile_test(source, args.compiler, "-Werror %s" % flags, None)
+    if success:
+        context.add_append_makefile_var(append_to, flags)
+        return
+
+    supported = []
+    for i in check_cflags:
```

Comment:
Leave `i` for numeric indices. It's better to be explicit here: use `flag`, `cflag` or something along these lines instead.